jib-core/src/main/java/com/google/cloud/tools/jib/docker/CliDockerClient.java

```diff
@@ -246,11 +246,20 @@ public DockerImageDetails inspect(ImageReference imageReference)
       throws IOException, InterruptedException {
     Process inspectProcess =
         docker("inspect", "-f", "{{json .}}", "--type", "image", imageReference.toString());
-    if (inspectProcess.waitFor() != 0) {
-      throw new IOException(
-          "'docker inspect' command failed with error: " + getStderrOutput(inspectProcess));
+
+    try (InputStreamReader stdout =
+        new InputStreamReader(inspectProcess.getInputStream(), StandardCharsets.UTF_8)) {
+
+      // CharStreams.toString() will block until the end of stream is reached.
+      String output = CharStreams.toString(stdout);
+
+      if (inspectProcess.waitFor() != 0) {
+        throw new IOException(
+            "'docker inspect' command failed with error: " + getStderrOutput(inspectProcess));
```

Comment:
@EternalWind  I think it would be safer to also exhaust StdErr stream the same way like StdOut to avoid the similar "being stuck" issue because of StdErr being full?

StdErr stream may have some data regardless whether process exited with success or error, so it should be read regardless of waitFor() result.